torch/_inductor/decomposition.py

```diff
@@ -177,6 +177,30 @@ def baddbmm(self, batch1, batch2, beta=1, alpha=1):
         self = self * beta
     return self + result
 
+@register_decomposition([aten.bmm])
+def bmm(self, batch2):
+    if self.size(1) == 1 and batch2.size(-1) == 1:
+        return torch.sum(self.squeeze(1) * batch2.squeeze(-1), dim=1, keepdim=True).unsqueeze(1)
+    return NotImplemented
+
+@register_decomposition([aten.addmm])
+def addmm(self, input1, input2, beta=1, alpha=1):
```

Comment:
Is there any problem with the default decomposition at https://github.com/pytorch/pytorch/blob/3d165dc3f3e1ba42c7512200b1b9c14d4556ec61/torch/_decomp/decompositions.py#L1177 so that you have to define it again here?

Reply:
Good point - this code went through several iterations. The initial code had both the addmm expansion and dot product and skinny matrix recognition in one function.  After a refactor, addmm now mirrors the code you pointed to.  I might test the present code without the addmm rule